### PR TITLE
[vscoq2] move doc_id inside execManager

### PR DESCRIPTION
**Pull request by gares**
feedback handling is still not correct

### CODE REVIEW

File: language-server/dm/executionManager.ml [hunk 1]
```diff
@@ -51,13 +51,20 @@ type state = {
   initial : Vernacstate.t;
   of_sentence : (sentence_state * feedback_message list) SM.t;
   options : options;
+  doc_id : int; (* unique number used to interface with Coq's Feedback *)
```
Comment (maximedenes) [inline language-server/dm/executionManager.ml:54]: Given the comment, it should be called `feedback_id` or something like that.

File: language-server/dm/executionManager.ml [hunk 2]
```diff
@@ -51,13 +51,20 @@ type state = {
...
 }
 
-let init vernac_state = {
-  initial = vernac_state;
-  of_sentence = SM.empty;
-  options = default_options;
-}
+let init =
+  let doc_id = ref (-1) in
```
Comment (maximedenes) [inline language-server/dm/executionManager.ml:58]: I found previous code (i.e. exposing a `fresh_xx_id` primitive) clearer than partial application with side effects, especially for OCaml non-experts.

File: language-server/dm/executionManager.ml (hunk 1 above)
Comment (gares) [reply language-server/dm/executionManager.ml:54]: A feedback message is `{ doc_id; route_id; span_id; contents }`. So to me `feedback_id` is even more confusing.

**Review by maximedenes — COMMENTED** [review]
After discussion, `doc_id` should eventually disappear in favor of a global state in exec manager (which can then use globally unique `sentence_id` to multiplex feedback).

So we can merge this or not, as is most convenient.

**Pull request merged**